WDE.DatabaseEditors/Models/DbTableField.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Newtonsoft.Json;
+using WDE.Common.Annotations;
+using WDE.DatabaseEditors.Data;
+using WDE.DatabaseEditors.History;
+
+namespace WDE.DatabaseEditors.Models
+{
+    public class DbTableField<T> : IDbTableField, INotifyPropertyChanged, IDbTableHistoryActionSource
+    {
+        // Constructor for serialization purpose
+        public DbTableField() { }
+        
+        public DbTableField(string fieldName, string inDbFieldName, bool isReadOnly, bool isModified, string valueType, bool isParameter,
+            T value)
+        {
+            FieldName = fieldName;
+            this.inDbFieldName = inDbFieldName;
+            IsReadOnly = isReadOnly;
+            this.isModified = isModified;
+            ValueType = valueType;
+            IsParameter = isParameter;
+            fieldValue = value;
+        }
+
+        public DbTableField(in DbEditorTableGroupFieldJson fieldDefinition)
+        {
+            FieldName = fieldDefinition.Name;
+            inDbFieldName = fieldDefinition.DbColumnName;
+            IsReadOnly = fieldDefinition.IsReadOnly;
+            isModified = false;
+            ValueType = fieldDefinition.ValueType;
+            IsParameter = fieldDefinition.ValueType.EndsWith("Parameter");
+            fieldValue = default;
+        }
+        
+        public DbTableField(in DbEditorTableGroupFieldJson fieldDefinition, T value)
+        {
+            FieldName = fieldDefinition.Name;
+            inDbFieldName = fieldDefinition.DbColumnName;
+            IsReadOnly = fieldDefinition.IsReadOnly;
+            isModified = false;
+            ValueType = fieldDefinition.ValueType;
+            IsParameter = fieldDefinition.ValueType.EndsWith("Parameter");
+            fieldValue = value;
+        }
+
+        public string FieldName { get; set; }
+        [JsonProperty]
+        private string inDbFieldName;
+        public bool IsReadOnly { get; set; }
+        [JsonProperty]
+        private bool isModified;
+
+        [JsonIgnore]
+        public bool IsModified
+        {
+            get => isModified;
+            set
+            {
+                isModified = value;
+                OnPropertyChanged(nameof(IsModified));
+            }
+        }
+        public string ValueType { get; set; }
+        public bool IsParameter { get; set; }
+        [JsonProperty]
+        private T fieldValue;
```

Comment:
Can we also keep here original value? It might be handy to check previous value (and this is something that normal database gui cannot do, so it is one more advantage of using the wow db editor).